frappe/handler.py

```diff
@@ -155,7 +155,7 @@ def upload_file():
 	frappe.local.uploaded_file = content
 	frappe.local.uploaded_filename = filename
 
-	if frappe.session.user == 'Guest' or (user and not user.has_desk_access()):
+	if (frappe.session.user == 'Guest' or (user and not user.has_desk_access())) and not file_url:
```

Comment:
If you make the `not file_url` check first, then the remaining expressions in the `and` clause won't be evaluated, which might save some calls.